File: solutions/swift/secret-handshake/2/Sources/SecretHandshake/SecretHandshake.swift
func commands(number: Int) -> [String] {
  var result = [String]()
  if number & 1 == 1 { result.append("wink") }
  if number & 2 == 2 { result.append("double blink") }
  if number & 4 == 4 { result.append("close your eyes") }
  if number & 8 == 8 { result.append("jump") }
  return number & 16 == 16 ? result.reversed() : result
}

File: solutions/haskell/secret-handshake/3/src/SecretHandshake.hs
module SecretHandshake (handshake) where

handshake :: Int -> [String]
handshake n
  | n >= 16 = reverse . handshake $ n - 16
  | otherwise = go n ["wink", "double blink", "close your eyes", "jump"]
  where
    go n' events
      | n' == 0 = []
      | odd n' = head events : go (n' `div` 2) (tail events)
      | otherwise = go (n' `div` 2) (tail events)
